proposals/11551-kubernetes-native-api/TestPlan.md

```diff
@@ -0,0 +1,87 @@
+| (KFP DB mode) Create a pipeline and 2 pipeline versions with different specs and run migrate script                                  |                                                                                                                                       | Pipeline and pipeline versions should match exactly except for ID?                                           |
+| (KFP DB mode) Create 2 pipelines and 1 pipeline version per pipeline spec and run migrate script                                     |                                                                                                                                       | Pipelines and pipeline versions should match exactly except for IDs?                                         |
+| (KFP DB mode) Create 2 pipeline and 2 pipeline versions with different specs and run migrate script                                  |                                                                                                                                       | Pipelines and pipeline versions should match exactly except for IDs?                                         |
+| (KFP DB mode) Create all pipelines and 1 pipeline version per pipeline spec under "data/pipeline_files/valid" and run migrate script | Run [Pipeline Upload API Tests](https://github.com/nsingla/kfp_pipelines/tree/pipeline_run_tests) without clean up and run migration  | Confirm that the migration script does not error out and is able to migrate all pipelines and their versions |
+| (K8s Mode) Create a pipeline in DB mode and switch to k8s mode and try creating the same pipeline without migration                  |                                                                                                                                       | Should disallow duplicate pipeline which hasn't been migrated?                                               |
+| (K8s Mode) Create a pipeline run in an experiment in DB mode and while the run is in progress switch to k8s mode                     |                                                                                                                                       | The run continue after migration                                                                             |
+| (K8s Mode) Create a pipeline recurring run in an experiment in DB mode and switch to k8s mode                                        |                                                                                                                                       | The run should still exist with the same cron settings                                                       |
```

Comment:
I think these will just work but I'm debating if we want to commit/guarantee this works considering it's just a one time migration and it could be tricky to automate. What do you think @HumairAK ?